mesonbuild/modules/pkgconfig.py

```diff
@@ -93,7 +93,17 @@ def _process_libs(self, libs, public):
                 if obj.found():
                     processed_libs += obj.get_link_args()
                     processed_cflags += obj.get_compile_args()
-            elif isinstance(obj, (build.SharedLibrary, build.StaticLibrary)):
+            elif isinstance(obj, build.SharedLibrary):
+                processed_libs.append(obj)
+            elif isinstance(obj, build.StaticLibrary):
+                # Due to a "feature" in pkgconfig, it leaks out private dependencies.
+                # Thus we will not add them to the pc file unless the target
+                # we are processing is a static library.
+                #
+                # This way (hopefully) "pkgconfig --libs --static foobar" works
+                # and "pkgconfig --cflags/--libs foobar" does not have any trace
+                # of dependencies that the build file creator has not explicitly
+                # added to the dependency list.
                 processed_libs.append(obj)
                 if public:
                     if not hasattr(obj, 'generated_pc'):
```

Comment:
I think this if should also be done for build.SharedLibrary. (i.e. obj.generated_pc = self.name if public)

Reply:
Doing that is _exactly_ what caused this bug.

Reply:
Why?
I think what causes this bug is one (or both) of the `self.add_priv_libs(` lines.

obj.generated_pc is only used if something else depends on this library, possibly even explicitly.